release/models/platform/openconfig-platform-chassis.yang

```diff
@@ -0,0 +1,79 @@
+module openconfig-platform-chassis {
+  yang-version "1";
+
+  namespace "http://openconfig.net/yang/platform/chassis";
+
+  prefix "oc-platform-chassis";
+
+  import openconfig-platform { prefix oc-platform; }
+  import openconfig-extensions { prefix oc-ext; }
+
+  organization "OpenConfig working group";
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module defines extensions to the OpenConfig platform model
+    that apply to chassis (CHASSIS) components.
+    These components are global to the device hardware package and
+    are not already part of another platform module and which
+    configuration or state is applicable.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2022-04-22" {
+    description
+      "Initial revision with chassis-id.";
+    reference "0.1.0";
+  }
+
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+
+  // extension statements
+
+  // feature statements
+
+  // identity statements
+
+  // typedef statements
+
+  // grouping statements
+
+  grouping chassis-config {
+    description
+      "Config that corresponds to the global hardware package";
+
+    leaf chassis-id {
+      type uint32;
+      description
+        "A unique identifier representing the global device.  This is
+        intended to be used as a root of a tree of component ids";
```

Comment:
Could you elaborate a bit more on the intention here?  Since this is tied to the `CHASSIS` component, in single chassis situations, I'll assume only ever a single value here per network-element but how would this be used in multi-chassis scenarios?  The component name would likely distinguish this as well.

What do you mean by "root of a tree of component ids" ?

Reply:
On behalf of Yang Zhang:  The intention here is for config-gen to have the ability to write a chassis-id number (uint32) into the switch, and for controller to read the value out. In a multi chassis situation, the config-gen will write different chassis-id numbers into different chassises. Similar to the /components/component/integrated-circuit/state/node-id which is for chip level id.

Reply:
These seems to conflict with the "id" that already exists at the top level: /components/component/state/id and there is a type in this state container as well that would be CHASSIS in this instance.  How will the "chassis-id" under /component/component/chassis/state be different?

Reply:
ACK to @cfyo here, I think that the intent here is specifically for P4RT implementations, moving this under the p4rt module, and giving it a name that differentiates it for this use seems like a good idea.

Reply:
In OC standard, the "/components/component/state/id" is a string type (http://ops.openconfig.net/branches/models/master/docs/openconfig-platform.html). 

In Arista switch, below is some state/id query results, basically echo the name in the path. 

`{
        "Path": "components/component[name=Ucd9032-3]/state/id",
        "values": {
          "components/component/state/id": "Ucd9032-3"
        }
      },
      {
        "Path": "components/component[name=SwitchAsic1]/state/id",
        "values": {
          "components/component/state/id": "SwitchAsic-1"
        }
      }`

The one we requested (config/chassis-id and state/chassis-id) here is different from the state/id in several aspects: 1) the type is uint32; 2) it is writable and readable; 3) it is used to indicate which node a physical trunk is associated to. Since a trunk can cross multiple chips, we have to associate the trunk with a chassis node (instead of the chip node) for controller applications, and assign the `chassis node id` (no such an id in the switch now) for trunk association here.

Reply:
Hi @ghostli123 - thanks for the response - a few things: 1) you could put an integer in the string and convert it  or propose modifying the type like making it a union like /components/component/type 2)  you could propose adding the id to the /components/component/config/id to make it writable 3) what other vendors have this same issue?

Reply:
Hi @cfyo, thanks for the suggestion! 

1) it depends on the vendor support. this makes the vendor support harder.

2) Same above. It would be easier for vendor to add a new leaf rather than updating the existing logic. (I do not have strong objection, but changing the existing OC path representation would take much longer time for vendor to adapt the change, compared with a new independent leaf).

3) I just have Arista box to check out. I assume that the implementation of other vendors follow the same standard regarding state/id path right now.

Reply:
What if we introduce a new, unique leaf for all components that is a user defined, configurable id.  This would make it consistent across all the components.  For example:  
/components/component/config/id-userdef
or
/components/component/config/description (this has precedent as a rw string in many other places)

@ghostli123 what do you think?  

If you think this might work for your use case, we need you to explain to the community in more detail what the use case is.   Why is it useful for the id to be a configurable integer?   Why does the read only, vendor defined id string not satisfy the use case?  

Maybe the above detail will help us choose a more intuitive name for the leaf.

If the use case is very unusual and not applicable, we may be better off doing a private augment that is not in openconfig.

Reply:
Hi @dplore, for sure, it works for me. If the leaf can be written, and there is a corresponding state node to read, for the chassis id, that is good enough for our use case. Thanks for the suggestion!

Reply:
moved to p4rt model so resolving this thread.  Please feel free to open comments on the new commit